Add guidelines for authorship of the MOOSE paper (#26321)
idaholab/moose: modules/doc/content/framework_development/authorship_guidelines.md

# MOOSE papers authorship guidelines

MOOSE is a collaborative effort and not only do we always welcome contributions, but also want to
credit them appropriately and in a sustainable manner. We do this by including our contributors
in the relevant journal article(s) that should be used by all, both within the community and externally,
to cite the MOOSE framework and its modules. Please see the [Citing.md] page for the current version
of these articles. These articles are published on a regular basis.

## MOOSE framework paper authorship guidelines

When citing MOOSE, whether when using it, using a derived application, or just referring to

> the MOOSE project, this is the article that should be cited. It is featured in the [Citing.md] page.

Reviewer: ```suggestion
the MOOSE project, the top-level MOOSE paper is the article that should be cited. It is featured in the [Citing MOOSE](Citing.md#MOOSE) page.
```

This paper is regularly replaced with either a new paper or an "update" paper. "Update" papers
refer to an update of the software and the paper, and can have a different authorship list as the original
paper.

In order to best acknowledge the current contributors to the framework, we strive to keep the authorship
of the MOOSE framework paper representative. To do that, we publish a summary of the new developments in the framework
and use the list of the principal contributors since the previous paper as the list of authors for the new paper.

The contributors are listed in two categories.

- Semi-permanent contributors are included in every new MOOSE framework paper by default. These contributors
  have written a very large part of the MOOSE framework and must be credited, at least until their contributions
  are overtaken by others.

The current criteria for semi-permanent authorship are listed below:

> - Top 10 line of codes in the `framework/` and other utility folders or

Reviewer: ```suggestion
- Top 10 contributor as measured by lines of code in the `framework/` and other utility folders or
```

> - Top 10 commits in the `framework/` folder

Reviewer: ```suggestion
- Top 10 contributor as measured by number of commits in the `framework/` folder
```

!alert note

> While it seems poor practice to recognize contribution by LOC or commit numbers, this criteria did not

Reviewer: ```suggestion
While it seems poor practice to recognize contribution by lines of code or commit numbers, this criteria did not
```

exist when these contributions were made, so the metric did not influence the coding style of these authors.

> These criteria may be reformulated in the future in terms of features.

Reviewer: ```suggestion
These criteria may be reformulated in the future in terms of contributed features and impact.
```

Just a suggestion.

- Transient contributors are included in the next MOOSE framework paper for the period(s) in which they contributed.
  These contributors have played an important role in MOOSE development since the previous paper was published.

The current guidelines for transient authorship are listed below:

- 1 major contribution or
- 4 minor contributions and significant involvement in MOOSE-related activities or
- 10 fixup contributions and significant involvement in MOOSE-related activities

> Contributions are only tallied for the *framework and supporting utilities*. Modules are deliberately excluded

Reviewer: ```suggestion
Contributions are only tallied for the *framework and supporting utilities*. Modules are deliberately excluded,
```

as developers of a module should publish a module paper and report on new developments outside of the MOOSE framework paper.
MOOSE-related activities are developments in the MOOSE modules and MOOSE applications.

A list of major, minor and fixup example contributions has been developed as guidelines for
classification and can be shared upon request. The classification of contributions is decided when
gathering the list of authors for the next paper.

!alert note
Contributors who do not meet the criteria for authorship in the paper will see their contributions from

> the past period considered for the next paper.

Reviewer: The wording here is a bit unclear. Does this mean that a contributor who doesn't meet the guidelines for the new update paper will have their contributions from the same period considered for the next paper (on top of new contributions in the intervening period)?

Consider revising.

Author: yes that would be my intention. 2 over 1 period then 2 over the next makes 4 for the next

!alert note
Code reviews and user support are currently not formalized in the criteria. Depending on the scale of the effort,
we may consider them at the level of minor and fixup contributions.

!alert note
The contribution guidelines may be scaled depending on the time elapsed between successive publications. The current
guidelines were developed for an 18 months time frame between two publications.

## MOOSE Modules paper authorship guidelines

Modules should strive to follow the same principles as the framework with regards to authorship inclusivity

> and representativity. However, most modules are not developed by the framework team and have their own policies.

Reviewer: I would add something about contacting the member of the CCB related to a given module (with a link to the CODEOWNERS file perhaps?)

Author: Codeowner s file s pretty rough git no?

Reviewer: A list of module directories next to GitHub handles? I don't think that's hard to parse for someone who is a contributor. Adding a link is just a suggestion, though.

Author: I dont think it matches each module's CCB that well. Reactor module for example, I would say most of their CCB is at ANL and not a codeowner because git still isnt developing this obvious need of non-maintainer codeowners

# MOOSE papers authorship guidelines

MOOSE is a collaborative effort and not only do we always welcome contributions, but also want to
credit them appropriately and in a sustainable manner. We do this by including our contributors
in the relevant journal article(s) that should be used by all, both within the community and externally,

> to cite the MOOSE framework and its modules. Please see the [Citing.md] page for the current versions

Author: ```suggestion
to cite the MOOSE framework and its modules. Please see the [citing.md] page for the current versions
```

> to cite the MOOSE framework and its modules. Please see the [citing.md] page for the current versions

Author: the "and its modules" part might be problematic. We want modules to be cited individually. 

Prefer
- cite the MOOSE framework.
- cite MOOSE
?

Reviewer 2: What about "cite the MOOSE framework and any modules used" ?

Author: I get what you mean that using a module you should still cite moose framework 

But citing a moose module externally from a publication that did not use it, you probably don't need to cite moose framework (though it does not hurt)

Author: oh nevermind this is a paragraph on both moose and its modules. this works

of these articles. These articles are published and updated on a regular basis.

## MOOSE framework paper authorship guidelines

When citing MOOSE, whether when using it, using a derived application, or just referring to

> the MOOSE project, the top-level MOOSE paper is the article that should be cited. It is featured in the [Citing MOOSE](Citing.md#MOOSE) page.

Author: ```suggestion
the MOOSE project, the top-level MOOSE paper is the article that should be cited. It is featured in the [Citing MOOSE](citing.md#MOOSE) page.
```

> the MOOSE project, the top-level MOOSE paper is the article that should be cited. It is featured in the [Citing MOOSE](citing.md#MOOSE) page.

Reviewer 2: ```suggestion
the MOOSE project, the top-level MOOSE framework paper should be cited. It is featured in the [Citing MOOSE](citing.md#MOOSE) page.
```

This paper is regularly replaced with either a new paper or an "update" paper. "Update" papers

> refer to a summary of changes related to a periodic update of the software from the timeframe of the previous "update" paper, and can have a different authorship list as the original

Reviewer 2: ```suggestion
refer to a summary of changes related to a periodic update of the software from the timeframe of the previous "update" paper, and can have a different authorship list than the original
```

paper.

In order to best acknowledge the current contributors to the framework, we strive to keep the authorship
of the MOOSE framework paper representative. To do that, we publish a summary of the new developments in the framework
and use the list of the principal contributors since the previous paper as the list of authors for the new paper.

The contributors are listed in two categories.

- Semi-permanent contributors are included in every new MOOSE framework paper by default. These contributors
  have written a very large part of the MOOSE framework and must be credited, at least until their contributions
  are overtaken by others.

The current criteria for semi-permanent authorship are listed below:

- Top 10 contributor as measured by lines of code in the `framework/` and other utility folders or
- Top 10 contributor as measured by number of commits in the `framework/` folder

!alert note
While it seems poor practice to recognize contribution by lines of code or commit numbers, this criteria did not
exist when these contributions were made, so the metric did not influence the coding style of these authors.

> These criteria may be reformulated in the future in terms of contributed features and impact.

Reviewer 2: I don't think this kind of note is necessary. If you want a note here, I might say that pull request reviewers will ask that duplicate/extraneous code be removed or refactored and that meaningless commits be squashed.

Reviewer 2: Or if you want to keep some of the content here that's ok, but I would consider mixing in my suggestions, and noting that we have always valued clean code and clean commit history (even if those values haven't 100% been honored all the time)

Author: re-worded. Maybe not everyone has to know that historically we did not use LOC / commits for authorship (though I suspect we did lol)

- Transient contributors are included in the next MOOSE framework paper for the period(s) in which they contributed.
  These contributors have played an important role in MOOSE development since the previous paper was published.

The current guidelines for transient authorship are listed below:

- 1 major contribution or
- 4 minor contributions and significant involvement in MOOSE-related activities or
- 10 fixup contributions and significant involvement in MOOSE-related activities

Contributions are only tallied for the *framework and supporting utilities*. Modules are deliberately excluded,
as developers of a module should publish a module paper and report on new developments outside of the MOOSE framework paper.

> MOOSE-related activities are developments in the MOOSE modules and MOOSE applications.

Reviewer 2: I think this sentence is confusing on the backs of the other two. It could be less confusing if you edit the bulleted list to say "1 major framework contribution", "4 minor framework contributions", etc.

Reviewer 2: If you do that, then I would delete the first sentence of this paragraph and then probably swap what are currently sentences two and three

A list of major, minor and fixup example contributions has been developed as guidelines for
classification and can be shared upon request. The classification of contributions is decided when
gathering the list of authors for the next paper.

!alert note
Contributors who do not meet the criteria for authorship in the paper will see their contributions from
the past period considered for the next paper.

!alert note
Code reviews and user support are currently not formalized in the criteria. Depending on the scale of the effort,
we may consider them at the level of minor and fixup contributions.

!alert note
The contribution guidelines may be scaled depending on the time elapsed between successive publications. The current
guidelines were developed for an 18 months time frame between two publications.

## MOOSE Modules paper authorship guidelines

Modules should strive to follow the same principles as the framework with regards to authorship inclusivity
and representativity. However, most modules are not developed by the framework team and have their own policies.

> Please contact the relevant module team for more information.

Reviewer 2: ```suggestion
Please contact the relevant module code owners for more information.
```

Reviewer 2: I think this makes it more tangible who one should contact

Author: I ll add a note. I dont consider code owners to be close enough to the team for each module

Author: see now

Reviewer 2: > I dont consider code owners to be close enough to the team for each module

Lol, what?! Then we don't have the correct code owners

Reviewer 2: I would hope that the code owners listed are good representatives of the module teams. If you're saying that not all module team members are listed as code owners, then I would agree with that.

Reviewer 2: I like what you wrote, so I'm resolving this convo

Author: Reactor team at ANL has no code owner.

Fluid properties is similarly a not full match of people in charge. 

I m sure tensor mechanics and phase field similarly have 1/2 the people we actually consider to be decision makers for those efforts in the codeowners file

Author: Ah i started typing too early. Yes you got what I mean
